primitives/src/pcs/univariate_kzg/mod.rs

```diff
@@ -910,4 +1158,91 @@ mod tests {
 
         Ok(())
     }
+
+    #[cfg(feature = "icicle")]
+    mod icicle {
+        use super::*;
+        use crate::{
+            icicle_deps::{curves::*, *},
+            pcs::univariate_kzg::icicle::GPUCommit,
+        };
+        use ark_ec::{short_weierstrass::Affine, CurveConfig};
+
+        fn test_gpu_e2e_template<E, C>() -> Result<(), PCSError>
+        where
+            C: IcicleCurve + MSM<C>,
+            C::ScalarField: ArkConvertible<ArkEquivalent = E::ScalarField>,
+            C::BaseField:
+                ArkConvertible<ArkEquivalent = <C::ArkSWConfig as CurveConfig>::BaseField>,
+            <C::ArkSWConfig as CurveConfig>::BaseField: PrimeField,
+            E: Pairing<G1Affine = Affine<<C as IcicleCurve>::ArkSWConfig>>,
+            UnivariateKzgPCS<E>: GPUCommit<E, C>,
+        {
+            let rng = &mut test_rng();
+            let stream = warmup_new_stream().unwrap();
+
+            let supported_degree = 2usize.pow(21);
+            let pp = UnivariateKzgPCS::<E>::gen_srs_for_testing(rng, supported_degree)?;
+            let (full_ck, _vk) = pp.trim(supported_degree)?;
+            let mut srs_on_gpu =
+                <UnivariateKzgPCS<E> as GPUCommit<E, C>>::load_prover_param_to_gpu(
+                    full_ck,
+                    supported_degree,
+                )?;
+
+            // testing on large degree for profiling
```

Comment:
I'm thinking about adding a profiling feature. But no action needed right now

Reply:
so you want something beyond arkworks' `print-trace`? 

i was on pause on this, maybe you can take a look if you find it interesting: https://github.com/EspressoSystems/jellyfish/pull/318

Reply:
Oh `print-trace` looks good enough. I mean running this part only when `print-trace` is enabled.

Reply:
i see, okay I can see if I can add this request in my next commit. stay tuned